src/ucp/api/ucp.h

```diff
@@ -1299,6 +1310,61 @@ ucs_status_t ucp_ep_flush(ucp_ep_h ep);
  * by a user, then the user is responsible for segmentation and subsequent
  * management.
  *
+ * <table>
+ * <caption id="ucp_mem_map_matrix">Matrix of behavior</caption>
+ * <tr><th>Parameter/Flag                  <th colspan="8">Value
+ * <tr><td>@ref UCP_MEM_MAP_NONBLOCK       <td colspan="8" align="center">
+ *                                         0/1 - the value affects only
+ *                                         registration/mapping phase </td>
+ * <tr><td>@ref UCP_MEM_MAP_ALLOCATE       <td align="center">0</td>
+ *                                         <td align="center">1</td>
+ *                                         <td align="center">0</td>
+ *                                         <td align="center">0</td>
+ *                                         <td align="center">1</td>
+ *                                         <td align="center">1</td>
+ *                                         <td align="center">0</td>
+ *                                         <td align="center">1</td>
+ * <tr><td>@ref UCP_MEM_MAP_FIXED          <td align="center">0</td>
+ *                                         <td align="center">0</td>
+ *                                         <td align="center">1</td>
+ *                                         <td align="center">0</td>
+ *                                         <td align="center">1</td>
+ *                                         <td align="center">0</td>
+ *                                         <td align="center">1</td>
+ *                                         <td align="center">1</td>
+ * <tr><td>@ref ucp_mem_map_params.address <td align="center">0</td>
+ *                                         <td align="center">0</td>
+ *                                         <td align="center">0</td>
+ *                                         <td align="center">defined</td>
+ *                                         <td align="center">0</td>
+ *                                         <td align="center">defined</td>
+ *                                         <td align="center">defined</td>
+ *                                         <td align="center">defined</td>
+ * <tr><td>Result                          <td>@ref anch_err "error"
+ *                                         <td>@ref anch_alloc_reg "alloc+register"
+ *                                         <td>@ref anch_err "error"
+ *                                         <td>@ref anch_reg "register"
+ *                                         <td>@ref anch_err "error"
+ *                                         <td>@ref anch_alloc_hint_reg "alloc+register,hint"
+ *                                         <td>@ref anch_err "error"
+ *                                         <td>@ref anch_alloc_fixed_reg "alloc+register,fixed"
+ * </table>
```

Comment:
Thanks for re-working the table and providing the 3 options.  I prefer the 2nd option; I think it is much cleaner in the pdf.  I have a couple of minor suggested modifications:
1. change the NONBLOCK column text from
    affects only\n registration/mapping\n phase
to
    only affects the\n register/map\n phase
this keeps the text from overflowing the column width

2. for the "address" column, modify the "1" value to "defined", as in table 3.